scenario/utils/relayOptimismMessage.ts

```diff
@@ -0,0 +1,81 @@
+import { DeploymentManager } from '../../plugins/deployment_manager';
+import { impersonateAddress } from '../../plugins/scenario/utils';
+import { setNextBaseFeeToZero, setNextBlockTimestamp } from './hreUtils';
+import { Event } from 'ethers';
+
+/*
+The Optimism relayer applies an offset to the message sender.
+
+applyL1ToL2Alias mimics the AddressAliasHelper.applyL1ToL2Alias fn that converts
+an L1 address to its offset, L2 equivalent.
+
+https://optimistic.etherscan.io/address/0x4200000000000000000000000000000000000007#code
+*/
+function applyL1ToL2Alias(address: string) {
+  const offset = BigInt('0x1111000000000000000000000000000000001111');
+  return `0x${(BigInt(address) + offset).toString(16)}`;
+}
+
+export default async function relayOptimismMessage(
+  governanceDeploymentManager: DeploymentManager,
+  bridgeDeploymentManager: DeploymentManager,
+) {
+  const EVENT_LISTENER_TIMEOUT = 60000;
+
+  const optimismL1CrossDomainMessenger = await governanceDeploymentManager.getContractOrThrow('optimismL1CrossDomainMessenger');
+  const bridgeReceiver = await bridgeDeploymentManager.getContractOrThrow('bridgeReceiver');
+  const l2CrossDomainMessenger = await bridgeDeploymentManager.getContractOrThrow('l2CrossDomainMessenger');
+
+  const { id: functionIdentifier, Interface } = governanceDeploymentManager.hre.ethers.utils;
+
+  // listen on events on the OptimismL1CrossDomainMessenger
+  const sentMessageListenerPromise = new Promise((resolve, reject) => {
+    const filter = {
+      address: optimismL1CrossDomainMessenger.address,
+      topics: [functionIdentifier('SentMessage(address,address,bytes,uint256,uint256)')]
+    };
+
+    governanceDeploymentManager.hre.ethers.provider.on(filter, (log) => {
+      resolve(log);
+    });
+
+    setTimeout(() => {
+      reject(new Error('OptimismL1CrossDomainMessenger.SentMessage event listener timed out'));
+    }, EVENT_LISTENER_TIMEOUT);
+  });
```

Comment:
We can use `ethers.provider.getLogs` to avoid the complexities of setting up our own listener with a timeout. Similar to how we do it for relaying Polygon messages.

Reply:
agreed; the Polygon version is much improved over this iteration. we should rework this to incorporate those improvements